File: coding-practice/50 Coding Challenges part 1/C#/problem22.cs
// Coding challenge #22 : Reverse an array
using System;
class problem22
{
    static int[] reverse_array(int[] input){
        int len = input.Length;
        int[] output = new int[len];
        for(int i=0; i<len; i++){
            output[i] = input[(len-i)-1];
        }

        return output;
    }

    static void Main(string[] args){
        int[] array = {1, 2, 3, 4, 5, 6};
        int[] reversed = new int[array.Length];
        reversed = reverse_array(array);
        Console.Write("The reversed version of ["+string.Join(", ",array)+"] is: ["+string.Join(", ",reversed)+"]");
    }
}

File: coding-practice/50 Coding Challenges part 1/JavaScript/scripts/problem22.js
// Coding challenge #22
// Reverse an array
function reverseArray(arr){
    var output = [];
    for(var i=arr.length-1; i>=0; i--){
        output.push(arr[i]);
    }

    return output;
}

var ar = [1, 2, 3];
var ar2 = reverseArray(ar);
console.log(ar2);